src/ipmanager/api/models.py

```diff
@@ -1 +1,55 @@
 # Create your models here.
+from django.db.models import TextField, IntegerField, CharField, \
+  BooleanField, CASCADE, ForeignKey, UniqueConstraint
+from django_extensions.db.models import TimeStampedModel
+from enum import Enum
+from ipaddress import IPv4Network
+from django.core.validators import RegexValidator
+import re
+from django.core.exceptions import ValidationError
+
+class Group(TimeStampedModel):
+  key = CharField(
+    max_length=32, 
+    unique=True, 
+    validators=[RegexValidator(regex=r"\A[a-z][a-z\d_-]*\Z", flags=re.IGNORECASE)]
+  )
+  name = CharField(max_length=128, unique=True)
+  description = TextField(blank=True)
+  notes = TextField(blank=True)
+  export = BooleanField(default=False)
+  
+def validate_ipv4_or_cidr_address(value):
+  try :
+    IPv4Network(value)
+    return True
+  except ValueError:
+    return False
+
+class IPRange(TimeStampedModel):
+  group_id = ForeignKey(Group, on_delete=CASCADE)
+  value = CharField(max_length=64, validators=[validate_ipv4_or_cidr_address])
+  class Meta:
+    constraints = [
+      UniqueConstraint(fields=['group_id', 'value'], name='unique_group_ip_pair')
+    ]   
+
+class RelationType(Enum):
```

Comment:
A more Django-ish way to do this is to use Django's own [enumeration types](https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types). This allows for simplifying the definition of the `relation` field:

```python
...
from django.db.models import IntegerChoices
...
class Relation(TimeStampedModel):
    class RelationType(IntegerChoices):
        INCLUSION = 0
        EXCLUSION = 1
    ...
    relation = IntegerField(choices=RelationType)
    ...
```